MekHQ/src/mekhq/gui/view/PlanetViewPanel.java

```diff
@@ -254,91 +132,96 @@ private void fillStats() {
+            panel.add(txtPosition, gbcText);
             ++ infoRow;
         }
         
-        lblJumpPoint.setName("lblJumpPoint"); // NOI18N
-        lblJumpPoint.setText(resourceMap.getString("lblJumpPoint1.text"));
-        gridBagConstraints = new GridBagConstraints();
-        gridBagConstraints.gridx = 0;
-        gridBagConstraints.gridy = infoRow;
-        gridBagConstraints.fill = GridBagConstraints.NONE;
-        gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
-        pnlStats.add(lblJumpPoint, gridBagConstraints);
-        
-        txtJumpPoint.setName("lblJumpPoint2"); // NOI18N
-        txtJumpPoint.setText(Double.toString(Math.round(100 * planet.getTimeToJumpPoint(1))/100.0) + " days");
+        //Time to Jump point
+        JLabel lblJumpPoint = new JLabel(resourceMap.getString("lblJumpPoint1.text"));
+        gbcLabel.gridy = infoRow;
+        panel.add(lblJumpPoint, gbcLabel);        
+        JTextArea txtJumpPoint = new JTextArea(Double.toString(Math.round(100 * planet.getTimeToJumpPoint(1))/100.0) + " days");
         txtJumpPoint.setEditable(false);
         txtJumpPoint.setLineWrap(true);
         txtJumpPoint.setWrapStyleWord(true);
-        gridBagConstraints = new GridBagConstraints();
-        gridBagConstraints.gridx = 1;
-        gridBagConstraints.gridy = infoRow;
-        gridBagConstraints.weightx = 0.5;
-        gridBagConstraints.insets = new Insets(0, 10, 0, 0);
-        gridBagConstraints.fill = GridBagConstraints.HORIZONTAL;
-        gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
-        pnlStats.add(txtJumpPoint, gridBagConstraints);
-        
+        gbcText.gridy = infoRow;
+        panel.add(txtJumpPoint, gbcText);
         ++ infoRow;
         
+        //Year length
+        if(null != planet.getYearLength()) {
+            JLabel lblYear = new JLabel(resourceMap.getString("lblYear1.text"));
+            gbcLabel.gridy = infoRow;
+            panel.add(lblYear, gbcLabel);        
+            JTextArea txtYear = new JTextArea(Double.toString(planet.getYearLength()) + " Terran years");
+            txtYear.setEditable(false);
+            txtYear.setLineWrap(true);
+            txtYear.setWrapStyleWord(true);
+            gbcText.gridy = infoRow;
+            panel.add(txtYear, gbcText);
+            ++ infoRow;
+        }
+        
+        //day length
+        if(null != planet.getDayLength()) {
+            JLabel lblDay = new JLabel(resourceMap.getString("lblDay1.text"));
+            gbcLabel.gridy = infoRow;
+            panel.add(lblDay, gbcLabel);        
+            JTextArea txtDay = new JTextArea(Double.toString(planet.getDayLength()) + " hours");
+            txtDay.setEditable(false);
+            txtDay.setLineWrap(true);
+            txtDay.setWrapStyleWord(true);
+            gbcText.gridy = infoRow;
+            panel.add(txtDay, gbcText);
+            ++ infoRow;
+        }
+        
+        /*
+           * TODO: fix satelllite information
```

Comment:
Hanging TODO

Reply:
Satellite information noew displays properly as of commit ec0c0c352719a7e582af968bbed61ee5696bfbc4.